Python/codegen.c

```diff
@@ -3723,6 +3723,97 @@ update_start_location_to_match_attr(compiler *c, location loc,
     return loc;
 }
 
+static int
+maybe_optimize_function_call(compiler *c, expr_ty e, jump_target_label end)
+{
+    asdl_expr_seq *args = e->v.Call.args;
+    asdl_keyword_seq *kwds = e->v.Call.keywords;
+    expr_ty func = e->v.Call.func;
+
+    if (! (func->kind == Name_kind &&
+           asdl_seq_LEN(args) == 1 &&
+           asdl_seq_LEN(kwds) == 0 &&
+           asdl_seq_GET(args, 0)->kind == GeneratorExp_kind))
+    {
+        return 0;
+    }
+
+    expr_ty generator_exp = asdl_seq_GET(args, 0);
+
+    if (asdl_seq_LEN(generator_exp->v.GeneratorExp.generators) != 1) {
+        return 0;
+    }
```

Comment:
I think this still works even if there's more than one loop in the generator, right?